[Neale] iP

nealetham opened the pull request:
# Duke+
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke+ frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use


All you need to do is,
1. download it from [here](https://github.com/nealetham/ip/releases/tag/A-Jar)
2. open the file in the terminal
```
java -jar Duke.jar
```
3. add your tasks

And it is **FREE!** 🤑 


## Features
- [x] Add tasks, deadlines, events
- [ ] Friendly user-interface (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the main method:
```ruby
public class Duke {
    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

nealetham committed:
Add Greet, Echo, Exit

nealetham committed:
Add Add, List

nealetham committed:
Add Mark as Done, Modify Main

nealetham committed:
Add Todo, Deadline, Event

nealetham committed:
Add testcase

nealetham committed:
Add DukeException, exception handling

nealetham committed:
Add Delete

nealetham committed:
Add Delete, edit documentation

nealetham committed:
Add save functionality

nealetham committed:
Add javadocs for methods in previous commit

nealetham committed:
Add type of deadline dates as LocalDateTime object

nealetham committed:
Removed comments

nealetham committed:
Merge branch 'branch-Level-8'

nealetham committed:
Add ui, storage, parser, tasklist

nealetham committed:
Add duke package

nealetham committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

nealetham committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

nealetham committed:
Add JUnit tests

nealetham committed:
Edit javadocs

nealetham committed:
Edit JavaDoc comments

nealetham committed:
Add find functionality

nealetham committed:
Edit indentations

nealetham committed:
Merge branch 'branch-Level-9'

nealetham committed:
Add checkstyle files, edit code to comply with checkstyle

Ssatu commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,50 @@
+     * @throws DukeException not thrown.
+     */
+    @Test
+    public void testCreateDeadline() throws DukeException {
```
I like how thoroughly you have tested the different classes! Perhaps a minor improvement could be to follow the three part format for test method names where relevant. e.g.  testCreateDeadline_testDeadline_dukeExceptionThrown()

Ssatu reviewed (commented):
Everything looks really good! Tough to find anything to comment on for Coding Standard. Good job!

nealetham committed:
Add GUI component to Duke

ReubenChay commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,45 @@
+     */
+    @Override
+    public String formatFileText() {
+        String s = String.format("E | %s | %s | %s\n", super.getStatusIcon(), this.description, this.at);
```
could use more descriptive name like output maybe?

ReubenChay commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,47 @@
+     * @return String of Duke's response.
+     * @throws DukeException when the input is not known.
+     */
+    public String parse(String input) throws DukeException {
```
really like this method of parsing as compared to my own, LGTM!

ReubenChay reviewed (commented):
Overall coding standard all good. Good extensive testing!

nealetham committed:
Add assertions to document important assumptions.

These assumptions include:
* All parameters from tasks of a previously loaded file are non-null
* All tasks saved into the file are correctly formatted
* Tasks are loaded into an empty TaskList
* Tasks saved into the file do not have an empty TaskList
* The time parameters for Events and Deadlines are not null
* User inputs are not empty blank Strings

nealetham committed:
Refactor codebase with minor edits

* Removed nested loops for run method in Duke.java
* Abstracted methods from Deadline constructor
* Removed nested loops from several methods in TaskList.java
* Abstracted methods from several methods in TaskList.java
* Abstracted methods from loadFile method in Storage.java

nealetham committed:
Merge pull request #1 from nealetham/branch-A-Assertions

Add assertions to document important assumptions.

nealetham committed:
Resolve merge conflicts

nealetham committed:
Merge pull request #2 from nealetham/branch-A-CodeQuality

Branch-A-CodeQuality

nealetham committed:
Setup GitHub CI

zylee348 commented on src/test/java/ParserTest.java:
```diff
@@ -1,8 +1,14 @@
-import duke.*;
-import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
```
good that you are not using wildcard imports

zylee348 reviewed (commented):
other wise, looks good

nealetham committed:
Add search tasks by partial keyword matching

As a separate functionality from the find keyword, match
allows users to search up tasks through partial keyword matching.
As opposed to find command  where there has to be a one-to-one
correspondence, match allows partial keyword search.

nealetham committed:
Add JavaDocs for several methods

nealetham committed:
Merge pull request #3 from nealetham/branch-BCD-Extension

Branch-BCD-Extension

nealetham committed:
Add user-guide, edit Ui images

nealetham committed:
Update hyperlink in README

nealetham committed:
Edit README, updated gradle for build

nealetham committed:
Update for build

nealetham committed:
Fix checkstyle, javadocs

nealetham committed:
Fix build

nealetham committed:
Remove CI I/O redirection tests
